runbot_travis2docker/models/runbot_branch.py

```diff
@@ -4,11 +4,40 @@
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
 import re
+import os
 
 import subprocess
 import requests
 
 from openerp import fields, models, api, tools
+try:
+    from functools import lru_cache
+except ImportError:
+    from backports.functools_lru_cache import lru_cache
```

Comment:
Add this package from manifest file

Reply:
Or use the lru cache of odoo in order to avoid install other library

Reply:
@moylop260 
The annotation "@tools.ormcache" only work over models

I try adding the annotation but the annotation fail

![image](https://user-images.githubusercontent.com/1387970/30402883-37c23042-98ad-11e7-9a90-f6d7e50689d1.png)

Nevertheless, the code I made avoid attaching the same key several times

What do you think about delete the code and the dependency "lru_cache" in favor of using my logic?

Reply:
We have the model runbot.branch and you could use that model to add this method.

> Nevertheless, the code I made avoid attaching the same key several times

The real issue here is read many times the same file for each project * branch * cron

Reply:
@moylop260 
Good point, I refactor the code to put the method "_ssh_keyscan" inside the class "runbot.branch"